record_batch.go

```diff
@@ -0,0 +1,204 @@
+package kafka
```

Comment:
Can we name this file `recordbatch.go` to match the naming style of the rest of the files?